### PR TITLE
Pull Request for branch queriesandtransactions that adds queries and transactions updates

### PR DESCRIPTION
**Pull request by jkanyi-web**
1. Added additional rows and an additional column to my animals table
2. Updated species column per animal
3. Made transactions, implemented savepoints, made updates based on certain conditions, used rollback, Deleted a row, made commits
4. Made queries that answer particular questions;

- How many animals are there?
- How many animals have never tried to escape?
- What is the average weight of animals?
- Who escapes the most, neutered or not neutered animals?
- What is the minimum and maximum weight of each type of animal?
- What is the average number of escape attempts per animal type of those born between 1990 and 2000?

**SCREENSHOTS**
Inside a transaction:
Delete all animals born after Jan 1st, 2022.
Create a savepoint for the transaction.
Update all animals' weight to be their weight multiplied by -1.
Rollback to the savepoint
Update all animals' weights that are negative to be their weight multiplied by -1.
Commit transaction

GUI3

Inside a transaction:
Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
Verify that changes were made.
Commit the transaction.
Verify that changes persist after commit.
GUI5

Inside a transaction update the animals table by setting the species column to unspecified. Verify that the change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.

unspecified

Now, take a deep breath and... Inside a transaction delete all records in the animals table, then roll back the transaction.
After the rollback verify if all records in the animals table still exists. After that, you can start breathing as usual ;)
GUI2

Video Screenshot of queries to answer the questions above

https://github.com/jkanyi-web/Vet-Clinic-DB/assets/107926388/a959f79c-7195-4485-b3aa-8314b7090bf5

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -4,4 +4,11 @@ INSERT INTO animals VALUES
     (1, 'Agumon', '2020-02-03', 0, 'true', 10.23),
     (2, 'Gabumon', '2018-11-15', 2, 'true', 8),
     (3, 'Pikachu', '2021-01-07', 1, 'false', 15.04),
-    (4, 'Devimon', '2017-05-12', 5, 'true', 11);
+    (4, 'Devimon', '2017-05-12', 5, 'true', 11),
+    (5, 'Charmander', '2020-02-08', 0, 'false', 11),
+    (6, 'Plantmon', '2021-11-15', 2, 'true', 5.7),
+    (7, 'Squirtle', '1993-04-02', 3, 'false', 12.13),
+    (8, 'Angemon', '2005-06-12', 1, 'true', 45),
```
Comment (1GORDON) [inline data.sql]: Great job @jkanyi-web 🤝

- [x] You have done well adding data to the animals' table. However, I would suggest you maintain negative values for the `weight_kg` columns for `Charmander`, `Plantmon`, `Squirtle`, and `Angemon` as they are provided in the project requirements.

File: queries.sql [hunk 2]
```diff
@@ -23,3 +23,71 @@ WHERE name != 'Gabumon';
 
 SELECT * FROM animals
 WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+
```
Comment (1GORDON) [inline queries.sql:26]: Great job @jkanyi-web 🤝

- [x] As stated in the project requirements, kindly set the species column to unspecified, then roll back the change. You may also verify that the change was made and rolled back.
transaction-1

File: queries.sql [hunk 3]
```diff
@@ -23,3 +23,71 @@ WHERE name != 'Gabumon';
...
+WHERE name LIKE '%mon';
+
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+
+SELECT name, species FROM animals;
+
+COMMIT;
+
```
Comment (1GORDON) [inline queries.sql:43]: Great job @jkanyi-web 🤝

- [ ]  As stated in the project requirements, kindly `verify that change persists after commit`.

**Review by 1GORDON — CHANGES_REQUESTED** [review]
##
## Hi @jkanyi-web ✋

Good job so far!

## Highlights!
- Created queries to answer questions in transactions ✔️
- Added data to the animals' table ✔️
- Added screenshots to the pull request description ✔️

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️

- [ ] As stated in the project requirements, kindly make sure your screenshots are numbered properly according to the query they represent.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql [hunk 4]
```diff
@@ -4,4 +4,11 @@ INSERT INTO animals VALUES
...
+    (9, 'Boarmon', '2005-06-07', 7, 'true', 20.4),	
+    (10, 'Blossom', '1998-10-13', 3, 'true', 17),	
+    (11, 'Ditto', '2022-05-14', 4, 'true', 22);
```
Comment (AbelG101) [inline data.sql]: - [ ] As mentioned previously please make sure to assign negative weights for `Charmander`, `Plantmon`, `Squirtle`, and `Angemon` as specified in the project requirements

File: queries.sql
Comment (AbelG101) [inline queries.sql:1]: - [ ] Once again as mentioned previously you seem to have missed the requirement for the transaction of setting the species column to unspecified make sure to verify the changes after and then roll back and verify once again. Please refer to this [previous comment](https://github.com/jkanyi-web/Vet-Clinic-DB/pull/2/files#r1371440215) for a reference.

File: queries.sql (hunk 3 above)
Comment (AbelG101) [inline queries.sql:43]: - [ ] As mentioned previously please make sure to verify that the changes have persisted after committing at line `37` you can easily do so by using the SQL statement you have at line `35`.

File: queries.sql
Comment (AbelG101) [inline queries.sql:1]: - [ ] Finally please make sure to commit all the changes you've made to the repository here as I noticed you didn't make any commits after the previous review.

**Review by AbelG101 — CHANGES_REQUESTED** [review]
# Hi @jkanyi-web 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- Correct use of Git Flow ✔
- Descriptive PR title with a good summary ✔

 ## Required Changes ♻️
- _Check the comments under the review._
- you just have to make a few changes and you're on your way 🚀.

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by binhussen — APPROVED** [review]
Hi @jkanyi-web , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

## Highlights
- Updated Animal Table✔️
- Well Structured Query ✔️
- Used Git flow ✔️
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**